sysboundary/setbyuser.cpp

```diff
@@ -270,6 +272,53 @@ namespace SBC {
       // No need to do anything in this function, as the propagators do not touch the distribution function   
    }
    
+   bool SetByUser::setBFromTemplate(const dccrg::Dccrg<SpatialCell,dccrg::Cartesian_Geometry>& mpiGrid,
+                                    FsGrid< std::array<Real, fsgrids::bfield::N_BFIELD>, 2> & perBGrid) {
+
+      std::array<bool,6> isThisCellOnAFace;
+      const std::array<int, 3> gridDims(perBGrid.getLocalSize());
+
+      for (int k=0; k<gridDims[2]; k++) {
+         for (int j=0; j<gridDims[1]; j++) {
+            for (int i=0; i<gridDims[0]; i++) {
+               const auto coords = perBGrid.getPhysicalCoords(i,j,k);
+               
+               // TODO: This code up to determineFace() should be in a separate function, it gets called in a lot of places.
+               // Shift to the center of the fsgrid cell
+               auto cellCenterCoords = coords;
+               cellCenterCoords[0] += 0.5 * perBGrid.DX;
+               cellCenterCoords[1] += 0.5 * perBGrid.DY;
+               cellCenterCoords[2] += 0.5 * perBGrid.DZ;
+
+               const auto refLvl = mpiGrid.get_refinement_level(mpiGrid.get_existing_cell(cellCenterCoords));
+
+               if(refLvl == -1) {
+                  cerr << "Error, could not get refinement level of remote DCCRG cell " << __FILE__ << " " << __LINE__ << endl;
+                  return false;
+               }
+
+               creal dx = P::dx_ini * pow(2,-refLvl);
+               creal dy = P::dy_ini * pow(2,-refLvl);
+               creal dz = P::dz_ini * pow(2,-refLvl);
+               
+               isThisCellOnAFace.fill(false);
+
+               determineFace(isThisCellOnAFace.data(), cellCenterCoords[0], cellCenterCoords[1], cellCenterCoords[2], dx, dy, dz);
+
+               for(uint iface=0; iface < 6; iface++) {
+                  if(facesToProcess[iface] && isThisCellOnAFace[iface]) {
+                     perBGrid.get(i,j,k)->at(fsgrids::bfield::PERBX) = templateB[iface][0];
+                     perBGrid.get(i,j,k)->at(fsgrids::bfield::PERBY) = templateB[iface][1];
+                     perBGrid.get(i,j,k)->at(fsgrids::bfield::PERBZ) = templateB[iface][2];
+                  }
```

Comment:
/!\ This should also have a break in order to preserve the precedence, see comment in the code below. /!\